packages/@ember/-internals/metal/lib/mixin.ts

```diff
@@ -575,6 +576,9 @@ export default class Mixin {
     this.ownerConstructor = undefined;
     this._without = undefined;
 
+    // Eagerly add INIT_FACTORY to avoid issues in DEBUG as a result of Object.seal(mixin)
+    this[INIT_FACTORY] = null;
```

Comment:
I'm still curious RE: debug guard here. Have you looked at that angle yet? It _seems_ like we should probably only need this when in debug so moving this down (into the if debug guard) would work just the same...

Reply:
Oh yes, you are correct.  Sorry about that.  I was confusing myself since the ultimate assignment is a production path.  But the `Object.seal` I believe only happens in DEBUG.  Fixed!